src/components/PolygonAddressInfo.js

```diff
@@ -0,0 +1,88 @@
+/* global IqonHash */
+/* global LoginFileConfig */
+/* global TemplateTags */
+/* global I18n */
+
+class PolygonAddressInfo { // eslint-disable-line no-unused-vars
+    /**
+     * @param {string} address
+     * @param {string} [label]
+     * @param {'none' | 'usdc' | 'unknown'} [logo = 'none']
+     */
+    constructor(address, label, logo = 'none') {
+        this._address = address;
+        this._label = label;
+        this._logo = logo;
+    }
+
+    /**
+     * Inserts this AddressInfo into $el overwriting the original content of $el.
+     * @param {HTMLElement} [$el]
+     * @returns {HTMLElement}
+     */
+    render($el) {
+        $el = $el || document.createElement('div');
+        $el.textContent = '';
+        $el.classList.add('polygon-address-info');
+
+        // Avatar or Icon
+        if (this._logo === 'none') {
+            const $avatar = document.createElement('div');
+            $avatar.classList.add('avatar');
+            if (this._label) {
+                $avatar.textContent = this._label.substring(0, 1).toUpperCase();
+                const color = IqonHash.getBackgroundColorIndex(this._label);
+                $avatar.classList.add(LoginFileConfig[color].className, 'initial');
+            } else {
+                $avatar.classList.add('unlabelled');
+            }
+            $el.appendChild($avatar);
+        } else if (this._logo === 'usdc') {
+            const $img = document.createElement('img');
+            $img.classList.add('logo');
+            $img.src = '../../assets/icons/usdc.svg';
+            $el.appendChild($img);
+        } else if (this._logo === 'unknown') {
+            const $img = document.createElement('img');
+            $img.classList.add('logo');
+            $img.src = '../../assets/icons/unknown.svg';
+            $el.appendChild($img);
+        }
```

Comment:
```suggestion
        } else if (this._logo === 'usdc' || this._logo === 'unknown') {
            const $img = document.createElement('img');
            $img.classList.add('logo');
            $img.src = `../../assets/icons/${this._logo}.svg`;
            $el.appendChild($img);
        }
```
Both cases can be combined.

Additionally, using a `background-image` instead of a separate image element could be considered, especially as the unknown icon already exists in the css via the `unlabelled` class.

Reply:
If the image already exists, then why join them? Just change the unknown case to use the `unlabelled` class. Combining these cases will just lead to us having to separate them again when things change.

Reply:
I disagree with this. This is not an optimization in my eyes.